src/core/components/hv-root/index.tsx

```diff
@@ -134,13 +134,14 @@ export default class Hyperview extends PureComponent<HvScreenProps.Props> {
     }
 
     // Re-render the modifications
-    opts.onUpdateCallbacks?.setNeedsLoad();
     opts.onUpdateCallbacks?.setState({
       doc: newRoot,
       elementError: null,
       error: null,
       url,
     });
+    // Force a reload regardless of state
+    opts.onUpdateCallbacks?.reload();
```

Comment:
Now I wonder if this call should be made as a callback of the `setState`, to match more closely previous implementation?
```suggestion
    }, () => {
      // Force a reload regardless of state
      opts.onUpdateCallbacks?.reload();
    });
```

Reply:
I'll have a look

Reply:
Updated the setState to support the functional call with callback.